qualtran/bloqs/multi_control_multi_target_pauli.py

```diff
@@ -148,3 +160,141 @@ def _ccpauli() -> MultiControlPauli:
     import_line='from qualtran.bloqs.multi_control_multi_target_pauli import MultiControlPauli',
     examples=(_ccpauli,),
 )
+
+
+@frozen
+class MultiControlX(Bloq):
+    r"""Implements multi-control, single-target X gate as a bloq using $n-2$ clean ancillas.
+
+    Args:
+        cvs: A tuple of control variable settings. Each entry specifies whether that
+            control line is a "positive" control (`cv[i]=1`) or a "negative" control (`cv[i]=0`).
+
+    Registers:
+        ctrls: An input register with n 1-bit controls corresponding to the size of the control
+            variable settings above.
+        x: A 1-bit input register bit-flipped based on the values in the ctrls register.
+
+    References:
+        [$C^n$NOT from n-2 Zeroed bits from Constructing Large Controlled Nots]
+        (https://algassert.com/circuits/2015/06/05/Constructing-Large-Controlled-Nots.html)
```

Comment:
Sorry we don't have a style guide for these references, but please use the actual title from the post as the link and then a period and then a note on what part of the reference we're referencing.

e.g.

[Constructing Large CNOTS](https:/...). Section title "n−2 Ancilla Bits", Figure titled CnNOT from n-2 zeroed bits.